File: offer/14/JPhoebe.py
#         self.val = x
#         self.next = None

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None


class Solution:
    def FindKthToTail(self, head, k):
        if head is None:
            return None
        if k == 0:
            return None
        stack = []
        while head is not None:
            stack.insert(0, head)
            head = head.next
        length = len(stack)
        if length < k:
            return None
        return stack[k - 1]


value = ListNode({1, 2, 3, 4, 5})
value.next = ListNode({2, 3, 4, 5})
value.next.next = ListNode({3, 4, 5})
value.next.next.next = ListNode({4, 5})
value.next.next.next.next = ListNode({5})

print(Solution().FindKthToTail(value, 1));

File: leetcode/206/jphoebe.java
package com.yunlsp.pay.sdktest.controller;


class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
    }
}

class Solution {

    ListNode listNode;
    public ListNode reverseList(ListNode head) {
        if (head.next != null){
            reverseList(head.next);
        }
        if (listNode == null){
            listNode = new ListNode(head.val);
        } else{
            listNode.next = new ListNode(head.val);
        }
        return listNode;
    }

    public ListNode reverseList2(ListNode head) {
        ListNode current, preview = head;
        head = head.next;
        while (head != null){
            ListNode nextTemp = head.next;
            current = head;
            current.next = preview;
            preview = current;
            head = nextTemp;
        }
        return head;
    }

}
